### PR TITLE
init scale test framework

### PR DESCRIPTION
**Pull request by allenlsy**
This PR is a starting point of scalability test.

It creates necessary functions to perform a simple test. It tests scaling up to 16 active gameservers from 1, 4, 16 standby servers respectively.

Uses openarena as testing game server to test scalability

* add functions `scale_up` and `scale_clear` to allocate new standing by servers. These method can also be used without `scale-test.sh` .
* add test cases regarding number of ~~standby~~ max servers

After running `scale-test.sh`, here is the result:

```
test 1: scale up to 16 servers from 1 standby server
Scale up time: 13s

test 2: scale up to 16 servers from 4 standby server
Scale up time: 34s

test 1: scale up to 16 servers from 16 standby server
Scale up time: 13s
```

Signed-off-by: Allen Leigh <allenlsy@gmail.com>

### CODE REVIEW

**Comment by dgkanatsios** [comment]
LGTM! YAML files reference the same GameServerBuild, right? Any way we can use one of them and then do `kubectl scale gameserverbuild <name> --replicas X` to do scaling?

File: tests/scalability/scale-test.sh [hunk 1]
```diff
@@ -2,21 +2,21 @@
 
 source ./util.sh
 
-GSID=85ffe8da-c82f-4035-86c5-9d2b5f42d6f7
+GSB_NAME=gameserverbuild-sample-openarena
 
 echo "test 1: scale up to 16 servers from 1 standby server"
 kubectl apply -f ./standby/1.yaml
```
Comment (dgkanatsios) [inline tests/scalability/scale-test.sh:8]: these kubectl apply commands are not valid anymore, right?

File: tests/scalability/scale-test.sh [hunk 2]
```diff
@@ -2,21 +2,21 @@
...
-scale_up $GSID 16
+scale_up $GSB_NAME 16
 kubectl delete gsb gameserverbuild-sample-openarena
```
Comment (dgkanatsios) [inline tests/scalability/scale-test.sh:10]: do we need to delete the build before scaling? we can downscale to zero if needed

Comment (allenlsy) [reply tests/scalability/scale-test.sh:10]: We can scale down to 0. Which one is more appropriate here?

File: tests/scalability/scale-test.sh (hunk 1 above)
Comment (allenlsy) [reply tests/scalability/scale-test.sh:8]: It creates the gsb with 1 initial standby and 16 max. I think we still need it.

File: tests/scalability/scale-test.sh (hunk 2 above)
Comment (dgkanatsios) [reply tests/scalability/scale-test.sh:10]: let's scale to zero, any chance we can wait till the actual standing by is zero?

File: tests/scalability/scale-test.sh (hunk 1 above)
Comment (dgkanatsios) [reply tests/scalability/scale-test.sh:8]: oh, sorry, files were not visible and I thought they were deleted. Do we need all three of them now?

Comment (allenlsy) [reply tests/scalability/scale-test.sh:8]: I think we need, but it shouldn't be different initial standby. We actually need to test the scale time for different max.

Comment (dgkanatsios) [reply tests/scalability/scale-test.sh:8]: max in all three files is currently the same, 16. Plus, we can send a JSON patch via kubectl and change it dynamically if needed. Not a big deal TBH, I wouldn't block this PR for it.

File: tests/scalability/scale-test.sh (hunk 2 above)
Comment (allenlsy) [reply tests/scalability/scale-test.sh]: Done

File: tests/scalability/scale-test.sh (hunk 1 above)
Comment (allenlsy) [reply tests/scalability/scale-test.sh]: Updated test cases to use max as the variant.

**Review by dgkanatsios — APPROVED** [review]
LGTM! Wanna take and slightly modify your PR description and put it in a README? After this we can merge

**Comment by allenlsy** [comment]
> LGTM! Wanna take and slightly modify your PR description and put it in a README? After this we can merge

Can you help update the PR description? I will add instructions to the readme, or a separate readme in this folder.

**Comment by dgkanatsios** [comment]
apologies, I wasn't clear. You don't have to modify the PR description itself, just whatever you copy and paste into the README. Hope it makes sense now!

**Pull request merged**